samples/openapi3/client/petstore/dart2/petstore_client_lib_fake/lib/api/fake_api.dart

```diff
@@ -215,7 +215,7 @@ class FakeApi {
     // At the time of writing this, `dart:convert` will throw an "Unexpected end of input"
     // FormatException when trying to decode an empty string.
     if (response.body != null && response.statusCode != HttpStatus.noContent) {
-      return apiClient.deserialize(_decodeBodyBytes(response), 'bool') as bool;
+      return response.body as bool;
```

Comment:
is this bad @kuhnroyal ? I guess it's kinda bad as it assumes there's no "malicious" input, but it does a job. Maybe this PR is doing too much already and I should revert some of the changes?

Reply:
I am not 100% sure, this needs to be tested, maybe a `json.decode` is still needed. That depends on the http library which I don't have much experience with.

Reply:
I think the PR is fine, but it is up to you if you want to split it up.

Reply:
>maybe a json.decode is still needed

I tests were failing with json.decode, this is testing body response which is NOT JSON or XML, but plain string/byte array.

This case is for  body response is `true` or `"I am string"`, which seems not about what openapi-spec is, but it's a part of http.